checker/src/main/java/org/checkerframework/checker/calledmethods/builder/LombokSupport.java

```diff
@@ -0,0 +1,204 @@
+
+        String methodName = element.getSimpleName().toString();
+
+        // make sure the method is toBuilder
+        if ("toBuilder".equals(methodName)) {
+            TypeElement enclosingElement = (TypeElement) element.getEnclosingElement();
+
+            if (ElementUtils.hasAnnotation(element, "lombok.Generated")
+                    || ElementUtils.hasAnnotation(enclosingElement, "lombok.Generated")) {
+                handleToBuilderType(returnType, enclosingElement);
+            }
+        }
+    }
+
+    /**
+     * Add, to a type, a CalledMethods annotation that states that all required setters have been
+     * called. The type can be the return type of toBuilder or the corresponding generated "copy"
+     * constructor.
+     *
+     * @param type type to update
+     * @param classElement corresponding AutoValue class
+     */
+    private void handleToBuilderType(AnnotatedTypeMirror type, Element classElement) {
+        List<String> requiredProperties = getLombokRequiredProperties(classElement);
+        AnnotationMirror calledMethodsAnno =
+                atypeFactory.createAccumulatorAnnotation(requiredProperties);
+        type.replaceAnnotation(calledMethodsAnno);
+    }
+
+    /**
+     * Computes the required properties of a @lombok.Builder class, i.e., the names of the fields
+     * with @lombok.NonNull annotations.
+     *
+     * @param lombokClassElement the class with the @lombok.Builder annotation
+     * @return a list of required property names
+     */
+    private List<String> getLombokRequiredProperties(final Element lombokClassElement) {
+        List<String> requiredPropertyNames = new ArrayList<>();
+        List<String> defaultedPropertyNames = new ArrayList<>();
+        for (Element member : lombokClassElement.getEnclosedElements()) {
+            if (member.getKind() == ElementKind.FIELD) {
+                for (AnnotationMirror anm :
+                        atypeFactory.getElementUtils().getAllAnnotationMirrors(member)) {
+                    if (NONNULL_ANNOTATIONS.contains(AnnotationUtils.annotationName(anm))) {
+                        requiredPropertyNames.add(member.getSimpleName().toString());
+                    }
+                }
+            } else if (member.getKind() == ElementKind.METHOD
+                    && ElementUtils.hasAnnotation(member, "lombok.Generated")) {
+                String methodName = member.getSimpleName().toString();
+                // Handle fields with @Builder.Default annotations.
+                // If a field foo has an @Builder.Default annotation, Lombok always generates a
+                // method called $default$foo.
+                if (methodName.startsWith("$default$")) {
+                    String propName = methodName.substring(9); // $default$ has 9 characters
+                    defaultedPropertyNames.add(propName);
+                }
+            } else if (member.getKind().isClass() && member.toString().endsWith("Builder")) {
```

Comment:
This is checking for the generated Builder class' element. It's a heuristic in that unless the `builderClassName` option to the `@Builder` annotation is passed, Lombok is guaranteed to generate an inner class with a name ending in `Builder` (e.g. `FooBuilder` for a `Foo` object). I've never seen anyone use that option in practice (though I'm sure some do), but it's not a soundness concern because the worst that happens is the code misses a singular/default annotation and expects you to set a field that's defaulted.

Reply:
I added a comment.